inst/staticexports/package.R

```diff
@@ -2,10 +2,97 @@ is_loaded <- function(x) {
   !is.null(.getNamespace(x))
 }
 
-is_installed <- function(package, version = NULL) {
-  installed <- nzchar(system.file(package = package))
+is_installed <- function(pkg, version = NULL) {
+  installed <- isNamespaceLoaded(pkg) || nzchar(system_file_cached(package = pkg))
   if (is.null(version)) {
     return(installed)
   }
-  installed && isTRUE(utils::packageVersion(package) >= version)
+  installed && isTRUE(get_package_version(pkg) >= version)
+}
+
+# Since I/O can be expensive, only utils::packageVersion() if the package isn't already loaded
+get_package_version <- function(pkg) {
+  ns <- .getNamespace(pkg)
+  if (is.null(ns)) {
+    utils::packageVersion(pkg)
+  } else {
+    as.package_version(ns$.__NAMESPACE__.$spec[["version"]])
+  }
+}
+
+
+# Borrowed from pkgload::shim_system.file, with some modifications.
+# Most notably, if the package isn't loaded via devtools, the package directory
+# lookup is cached. Also, to keep the implementation simple, it doesn't support
+# specification of lib.loc or mustWork
```

Comment:
```suggestion
# Borrowed from pkgload::shim_system.file, with some modifications. This behaves
# like `system.file()`, except that (1) for packages loaded with
# `devtools::load_all()`, it will return the path to files in the package's
# inst/ directory, and (2) for other packages, the directory lookup is cached.
# Also, to keep the implementation simple, it doesn't support specification of
# lib.loc or mustWork.
```